api/src/pcapi/core/educational/api/offer.py

```diff
@@ -604,6 +608,114 @@ def attach_image(
         db.session.commit()
 
 
+def update_collective_offer(
+    offer_id: int,
+    new_values: dict,
+) -> None:
+    offer_to_update = educational_models.CollectiveOffer.query.filter(
+        educational_models.CollectiveOffer.id == offer_id
+    ).first()
+
+    if "venueId" in new_values.keys():  # if offer_to_update.venueId in new_values["venueId"]:
```

Comment:
plusieur petites remarques: 
1) le `.keys()` ne sert a rien ici:
```suggestion
    if "venueId" in new_values:  # if offer_to_update.venueId in new_values["venueId"]:
```
est tres bien et plus pythonique
2) petite optimisation pour éviter de faire le calcul si on nous passe l'argument mais que la valeur n'a pas changé
```suggestion
    if "venueId" in new_values and new_values["venueId"] != offer_to_update.venueId:
```

Reply:
hummm je ne peux etre d'accord dans le sens ou cette modification du code va entrainer un bug.

En effet, si je place 
```python 
and new_values["venueId"] != offer_to_update.venueId:
```

alors il y aura une incompatibilité car new_values["venueId"] != offer_to_update.venueId est supposé raiser une erreur.